src/services/validationService.ts

```diff
@@ -0,0 +1,155 @@
+import {
+  ICommon,
+  ILimits,
+  IRules,
+  IValidationResponse,
+  IQuery,
+  IString,
+} from "../models";
+import constants from "../resources/constants.json";
+
+class ValidationService {
+  public static validateNativeQuery(
+    query: IQuery,
+    limits: ILimits
+  ): IValidationResponse {
+    if (limits != undefined && limits) {
+      this.validateCommonRules(query, limits.common);
+      switch (query.queryType) {
+        case "groupBy":
+          return this.validateQueryTypes(query, limits.queryRules.groupBy);
+        case "topN":
+          return this.validateQueryTypes(query, limits.queryRules.topN);
+        case "scan":
+          return this.validateQueryTypes(query, limits.queryRules.scan);
+        case "search":
+          return this.validateQueryTypes(query, limits.queryRules.search);
+        case "timeseries":
+          return this.validateQueryTypes(query, limits.queryRules.timeseries);
+        case "timeBoundary":
+          return this.validateQueryTypes(query, limits.queryRules.timeBoundary);
+        default:
+          return { isValid: true };
+      }
+    } else {
+      return { isValid: true };
+    }
+  }
+
+  private static validateQueryTypes(
+    query: IQuery,
+    queryRules: IRules
+  ): IValidationResponse {
+    return this.validateDateRange(query.intervals, queryRules.max_date_range);
+  }
+
+  private static validateDateRange(
+    dateRange: string[] | string = "",
+    allowedRange: Number = 0
+  ): IValidationResponse {
+    if (dateRange) {
+      const extracted = Array.isArray(dateRange)
+        ? dateRange[0].split("/")
+        : dateRange.toString().split("/");
+      const fromDate = new Date(extracted[0]).getTime();
+      const toDate = new Date(extracted[1]).getTime();
+      const diffInDays = (toDate - fromDate) / (1000 * 3600 * 24);
+      if (diffInDays > allowedRange) {
+        return {
+          errorMessage: constants.invalidDateRange.replace(
+            "${allowedRange}",
+            allowedRange.toString()
+          ),
+          isValid: false,
+        };
+      } else {
+        return { isValid: true };
+      }
+    } else {
+      return {
+        errorMessage: constants.noDataRange,
+        isValid: false,
+      };
+    }
+  }
+
+  private static validateCommonRules(query: IQuery, limits: ICommon): void {
+    if (query.threshold) {
+      query.threshold =
```

Comment:
method name says validate date range but logic is validating and updating the global variables and its return type is "avoid". I assumed the validation will return whether it obeys the rules or not.